src/app/workflow-editor/object-inspector/io-inspector/workflow-io-inspector.component.ts

```diff
@@ -324,16 +328,15 @@ export class WorkflowIOInspectorComponent extends DirectiveBase implements OnIni
     }
 
     labelUpdate(ev: FocusEvent) {
-        let val = (<HTMLInputElement>ev.srcElement).value;
+        const el = <HTMLInputElement>ev.srcElement;
+        let val = el.value;
 
         if (!val) {
-            val = (<HTMLInputElement>ev.srcElement).value = this.port.label;
-        }
-
-        if (this.port.label !== val) {
+            this.port.label = undefined;
+        } else if (this.port.label !== val) {
```

Comment:
This extra check is unnecessary, because even in the case where `this.port.label === val`, the graph will still be redrawn. The logic might be cleaner like this:

```javascript
if (val === this.port.label) {
    return;
}

this.port.label = val ? val : undefined;
this.graph.redraw();
```